### PR TITLE
Alpine based + multi-arch

### PR DESCRIPTION
**Pull request by apaillier-ledger**
Complete revamp of the app builder :tada: 

* Now based on Alpine Linux instead of Ubuntu (~600 MB less)
* Now gets all its packages straight from the repositories instead of manually downloading and installing some of them
* Thanks to the previous point, the Dockerfile becomes architecture-agnostic and the container is now generated for both linux/amd64 & linux/arm64 (simpler than #23)
* New base container suffixed with -lite, without the Rust environment so that apps that only use C can enjoy an even smaller container (extra ~1.3 GB less)
* New CI that builds both containers
* pytest removed from the container to keep it strictly focused on app-building

### CODE REVIEW

**Review by xchapron-ledger — COMMENTED** [review]
With this changes, neither `LLVM` nor `gcc-arm-none-eabi` version are specified.
They were previously specified which was allowing to be sure that it doesn't change without explicit user action.

Does it matter?
This was also allowing to easily check the version of a specific docker image without having to run it.

**Comment by apaillier-ledger** [comment]
> With this changes, neither `LLVM` nor `gcc-arm-none-eabi` version are specified. They were previously specified which was allowing to be sure that it doesn't change without explicit user action.

You are right they are not specified anymore but what makes this container always build with the same versions of such packages is that we're using a stable version of Alpine as a base container (3.15 here).
The only things that might change from one container build to another are [bugfixes backported to some packages](https://wiki.alpinelinux.org/wiki/Alpine_Linux:FAQ#What_is_the_difference_between_edge_and_stable_releases.3F).

**Comment by jibeee** [comment]
It seems that Alpine has `nano.specs` for newlib, which was the main reason why the ARM embedded toolchain was used on Ubuntu. That's great.

Why do you use Alpine 3.15, and not 3.16? That forces to stay with LLVM 12, while 3.16 would upgrade to LLVM 13.

**Comment by jibeee** [comment]
Probably some apps won't build (I have Tron in mind), as you removed some packages (protoc, for example). Build jobs for these apps need to be adapted in consequence.

**Comment by apaillier-ledger** [comment]
@jibeee 

> It seems that Alpine has `nano.specs` for newlib, which was the main reason why the ARM embedded toolchain was used on Ubuntu. That's great.
> 
> Why do you use Alpine 3.15, and not 3.16? That forces to stay with LLVM 12, while 3.16 would upgrade to LLVM 13.

Simply because the current environment that actually builds apps and pushes them onto the app store does not currently use this Docker container and still uses LLVM 12. Upgrading this container to a more recent version could create a situation where an app can be successfully built with this container but fails with the deployment environment. Keeping versions the same on both sides prevents that from happening.

> Probably some apps won't build (I have Tron in mind), as you removed some packages (protoc, for example). Build jobs for these apps need to be adapted in consequence.

Nice catch, I just added it back. :+1:

File: lite/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        cmake \
+        make \
+        doxygen \
+        git \
+        python3 \
+        py3-pip \
+        bash \
+        gcc-arm-none-eabi \
+        newlib-arm-none-eabi \
+        protoc
```
Comment (jibeee) [inline lite/Dockerfile:20]: Please sort package names [alphabetically](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments).

File: lite/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+# These packages contain shared libraries which will be needed at runtime
+RUN apk add \
+        libjpeg \
+        zlib \
+        eudev \
+        libusb
+
+# Python packages building dependencies, can be removed afterwards
+ARG PYTHON_BUILD_DEPS=python3-dev,musl-dev,libusb-dev,eudev-dev,linux-headers,zlib-dev,jpeg-dev
```
Comment (jibeee) [inline lite/Dockerfile:33]: Same here

**Review by jibeee — CHANGES_REQUESTED** [review]

**Comment by jibeee** [comment]
I like these changes, but I really fear it will break many jobs.

For example, you removed lcov. Aren't there any app using lcov for unit tests?

File: lite/Dockerfile (hunk 1 above)
Comment (apaillier-ledger) [reply lite/Dockerfile]: Done in [fe172af](https://github.com/LedgerHQ/ledger-app-builder/pull/36/commits/fe172afb8eadcb3519017aec39cf3e8784bb0101)

File: lite/Dockerfile (hunk 2 above)
Comment (apaillier-ledger) [reply lite/Dockerfile]: Done in [fe172af](https://github.com/LedgerHQ/ledger-app-builder/pull/36/commits/fe172afb8eadcb3519017aec39cf3e8784bb0101)

**Review by jibeee — APPROVED** [review]

**Comment by jibeee** [comment]
Would be good to have @Saltari 's feedback on that PR.

**Review by Saltari — APPROVED** [review]

**Pull request merged**